### PR TITLE
[DEVOPS-74]  Deploy EKG->statsd->Datadog stats

### PR DESCRIPTION
**Pull request by deepfire**
This wires things up so that https://app.datadoghq.com/metric/explorer?live=true&page=0&is_auto=false&from_ts=1497925423905&to_ts=1497929023905&tile_size=m&exp_metric=rts.gc.gc_wall_ms&exp_scope=&exp_agg=avg&exp_row_type=metric can be observed : -)

### CODE REVIEW

File: modules/cardano-node.nix [hunk 1]
```diff
@@ -44,6 +44,7 @@ let
        then "--bitcoin-distr \"${distributionParam}\""
        else "--flat-distr \"${distributionParam}\""))
     (optionalString cfg.jsonLog "--json-log ${stateDir}/jsonLog.json")
+    (optionalString (cfg.ekgSink != null) "--metrics +RTS -T -RTS --statsd-server ${cfg.ekgSink}")
```
Comment (domenkozar) [inline modules/cardano-node.nix]: @dcoutts @dcoutts-2fa @kantp should we enable `-T`?

File: modules/cardano-node-aws.nix [hunk 2]
```diff
@@ -19,6 +19,7 @@ testIndex: region:
         );
         publicIP = if options.networking.publicIPv4.isDefined then config.networking.publicIPv4 else null;
         privateIP = if options.networking.privateIPv4.isDefined then config.networking.privateIPv4 else "0.0.0.0";
+        ekgSink = "127.0.0.1:8125";
```
Comment (domenkozar) [inline modules/cardano-node-aws.nix]: I'd keep names consistent with cardano-sl and name it `statsdServer`

File: modules/cardano-node.nix (hunk 1 above)
Comment (dcoutts) [inline modules/cardano-node.nix]: Yes, we should use `-T`, either when we invoke (as this patch does) but probably makes sense to bake it into the binary at link time by specifying `-with-rtsopts=-T` in the .cabal file.

The `-T` is needed to tell the RTS to collect GC stats, which is one of the things we report via ekg. Without that the GC numbers will all be 0.

Comment (domenkozar) [inline modules/cardano-node.nix]: @avieth can you incorporate this into your PR at https://github.com/input-output-hk/cardano-sl/pull/368

**Comment by deepfire** [comment]
@domenkozar, the testing changes are consciously left in for a subsequent undoing -- they are in the top commit.  This is to facilitate testing of new changes.

If you think it would make sense, I'll be submitting future PR's with the testing commit undone -- that'll involve slightly more git bureaucracy on my side (esp. if there are review changes to implement), but it can be done.

Maybe it's kind of impolite of me to make you see these, I don't know.. : -)

File: modules/cardano-node-aws.nix (hunk 2 above)
Comment (deepfire) [inline modules/cardano-node-aws.nix]: Done

**Comment by deepfire** [comment]
@domenkozar, GitHub claims this PR is merged -- but I can't find `d46c7fbb2283551933901fc19e5c0b9b088fd42c` anywhere in the repository.

**Comment by deepfire** [comment]
Seems like at least https://github.com/input-output-hk/iohk-nixops/commit/d46c7fbb2283551933901fc19e5c0b9b088fd42c#diff-9769191803064c6543457a521474e83bR22 never made it in.

```
staging@iohk-infra:~/staging$ ag statsdServer
staging@iohk-infra:~/staging$
```

**Comment by domenkozar** [comment]
That's a shame, I have most probably force pushed it. Exactly why we need DEVOPS-173.

**Comment by domenkozar** [comment]
Pushed to staging.

**Pull request merged**